maas-api.yaml

```diff
@@ -1,555 +1,677 @@
+        - Coverage
+      summary: All covered regions
       parameters:
-        - name: input
-          in: body
+        - name: If-None-Match
+          in: header
+          required: false
+          description: ETag (e.g., hash code) of your last response. If this is supplied the response might return a 304, if the data has not changed.
           schema:
-            $ref: '#/definitions/RegionsInput'
+            type: string
+      description:
+        Lists available regions and available transport modes. Provide optional hash code to only return output if the data has changed.
       responses:
         200:
           description: Available regions and modes
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/RegionsResponse'
+          headers:
+            ETag:
+              description: ETag (e.g., hash code) of the content of the response. You can pass this as an input header, if you only want to get content in the response if the data has changed.
+              schema:
+                type: string            
+        304:
+          description: No change, i.e., If-None-Match matches ETag
+  
+  /regions/contain:
+    get:
+      tags:
+        - Coverage
+      summary: Coverage check by point(s)
+      parameters:
+        - name: points
+          in: query
+          required: true
+          description: Array of points (string of lat-long pairs, separated by a comma) to check for coverage. Result will indicate whether *all* of the provided points are covered.
           schema:
-            $ref: '#/definitions/RegionsResponse'
+            type: array
+            items:
+              type: string
+              description: Latitude-longitude pair as a string, separated by a comma
+      responses:
+        200:
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  covered:
+                    type: boolean
+                  regionIDs:
+                    type: array
+                    items:
+                      type: string
 
-  /regionInfo.json:
-    post:
-      tags:
-        - Configuration
-      summary: TSPs per region
-      description:
-        Retrieves basic information about covered transport service providers for the specified regions.
+  /regions/{id}:
+    get:
+      tags: 
+        - Coverage
+      summary: Covered region
       parameters:
-        - name: input
-          in: body
+        - name: id
+          in: path
+          required: true
           schema:
-            $ref: '#/definitions/RegionInfoInput'
+            type: string
       responses:
         200:
           description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Region'
+
+  /regions/{id}/details:
```

Comment:
can we merge this into `/regions/{id}` and always return all details?

Reply:
From an implementation side, rather not:

1. `details` contains information that's possibly dynamic. Just looking at our backend, having all that available for all regions for a single API call is not trivial. 
2. `/regions` output in our case is already very large, also adding all those details for every regions makes the output massive.

The idea here is to have `/regions` return high level coverage details ("what regions do you support, with what *modes* in a region") while `details` include much more granular information of what *data* is available in each region.

Reply:
i understand `/regions` returns a summary for all/multiple regions.  i just don't get why `/regions/{id}` would return a summary.

Reply:
Done. One endpoint less. Hooray.